src/test/java/org/openrewrite/java/dropwizard/method/RemoveUnnecessaryOverrideTest.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.dropwizard.method;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class RemoveUnnecessaryOverrideTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipes(new RemoveUnnecessaryOverride(false));
+    }
+
+    @Test
+    void removesUnnecessaryOverride() {
+        rewriteRun(
+          java(
+            """
+              class BaseClass {
+                  void baseMethod() {}
+              }
+              """),
+          java(
+            """
+              class TestClass extends BaseClass {
+                  @Override
+                  void nonExistentMethod() {}
+              }
+              """,
+            """
+              class TestClass extends BaseClass {
+
+                  void nonExistentMethod() {
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @DocumentExample
+    @Test
+    void configuresRenderCommandArguments() {
+        rewriteRun(
+          spec ->
+            spec.parser(JavaParser.fromJavaVersion().classpath("argparse4j", "spring-boot")),
+          java(
+            """
+              import net.sourceforge.argparse4j.impl.Arguments;
+              import net.sourceforge.argparse4j.inf.Subparser;
+              import org.springframework.boot.CommandLineRunner;
+
+              public class RenderCommand implements CommandLineRunner {
+                  @Override
+                  public void configure(Subparser subparser) {
```

Comment:
This is a bit of an odd mix of classes used here: it seems to test that halfway through the migration we can clear out an override. Would it not make more sens to have a unit test for the full declarative recipe verify that a fully Dropwizard input is converted to the expected Spring Boot equivalent?